python/tests/unit/arcticdb/version_store/test_read_index.py

```diff
@@ -0,0 +1,413 @@
+"""
+Copyright 2024 Man Group Operations Limited
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt. As of the
+Change Date specified in that file, in accordance with the Business Source License, use of this software will be
+governed by the Apache License, version 2.0.
+"""
+
+import pandas as pd
+import numpy as np
+import arcticdb
+import pytest
+from datetime import datetime
+from functools import reduce
+from packaging.version import Version
+from arcticdb.encoding_version import EncodingVersion
+from arcticdb.util._versions import PANDAS_VERSION
+from arcticdb_ext.exceptions import InternalException
+from arcticdb.util.test import  CustomThing, TestCustomNormalizer
+from arcticdb.version_store._custom_normalizers import register_normalizer, clear_registered_normalizers
+from arcticdb.options import LibraryOptions
+from arcticdb import ReadRequest
+from arcticdb.util.test import assert_frame_equal
+
+@pytest.fixture(
+    scope="function",
+    params=(
+        pd.RangeIndex(start=0, stop=10),
+        pd.RangeIndex(start=0, stop=10, step=2),
+        pd.RangeIndex(start=5, stop=25, step=5),
+        pd.date_range(start="01/01/2024",end="01/10/2024"),
+        pd.MultiIndex.from_arrays(
+            [pd.date_range(start="01/01/2024", end="01/10/2024"), pd.RangeIndex(start=0, stop=10)],
+            names=["datetime", "level"]
+        )
+    )
+)
+def index(request):
+    yield request.param
+
+class TestBasicReadIndex:
+
+    @pytest.mark.parametrize("dynamic_schema", [False, True])
+    def test_read_index_columns(self, lmdb_storage, index, lib_name, dynamic_schema):
+        ac = lmdb_storage.create_arctic()
+        lib = ac.create_library(lib_name, LibraryOptions(dynamic_schema=dynamic_schema))
+        df = pd.DataFrame({"col": range(0, len(index))}, index=index)
+        lib.write("sym", df)
+        result = lib.read("sym", columns=[])
+        assert isinstance(result, arcticdb.VersionedItem)
+        assert result.symbol == "sym"
+        assert result.version == 0
```

Comment:
We can omit the 3 asserts on lines 49-51 in the other tests, they add a lot of verbosity without adding much value